src/definition/populate_test.go

```diff
@@ -516,15 +524,37 @@ func TestIntegrationProtocol2PopulateFunc_PopulateOnlySpecifiedGroups(t *testing
 	)(groups, withGeneratorSpec)
 	assert.True(t, populated)
 	assert.Empty(t, errs)
-	assert.Contains(t, integration.Data, &expectedEntityData1)
-	assert.Contains(t, integration.Data, &expectedEntityData2)
-	assert.Contains(t, integration.Data, &expectedEntityData3)
-	assert.Len(t, integration.Data, 3)
+	assert.Len(t, dummyIntgr.Entities, 3)
+
+	compareIgnoreFields := cmpopts.IgnoreUnexported(integration.Entity{}, metric.Set{}, inventory.Inventory{})
+
+	// Search for missing entities using cmp
+	for _, ti := range []*integration.Entity{
+		expectedEntityData1, expectedEntityData2, expectedEntityData3,
+	} {
+		found := false
+		diff := ""
+		for _, presentEntity := range dummyIntgr.Entities {
+			if cmp.Equal(presentEntity, ti, compareIgnoreFields) {
+				found = true
+				break
+			}
+			// Store "closest match" for better debuggability
+			curDiff := cmp.Diff(presentEntity, ti, compareIgnoreFields)
+			if len(diff) == 0 || len(curDiff) < len(diff) {
+				diff = curDiff
+			}
+		}
+
+		if !found {
+			t.Fatalf("Entity list does not contain %s. Closest match: %s", ti.Metadata.Name, diff)
+		}
+	}
 }
 
 func TestIntegrationProtocol2PopulateFunc_EntityTypeGeneratorFuncWithError(t *testing.T) {
 	generatorWithError := func(_ string, _ string, _ RawGroups, _ string) (string, error) {
-		return "", errors.New("error generating entity type")
```

Comment:
In such case generally using `errors.New()` is permissible, as using `fmt.Errorf` without wrapping is "taking more functionality than needed". On the other hand, that require importing another package for no very good reason.

Though I personally accept using `fmt.Errorf` without formatting.